magento/magento2
# Fix handling of mixins for modules with no dependencies defined

### Description (*)
This PR addresses an issue that was outlined in https://github.com/magento/magento2/pull/25587#issuecomment-611074021 and investigated by me afterwards.

This is a regression I have introduced when refactoring mixins module, but it applies only in two cases which are not present in the Magento itself:
```javascript
// Module defined as anything else then a function with dependencies skipped.
define('my-module', {
  foo: 'bar'
});
// Module defined with dependencies passed as "null".
define('my-module', null, {
  foo: 'bar'
});
```
which both results in special case where second element of RequireJS's `defQueue` is `null` and made the script throw an error when somebody required such module.

### Manual testing scenarios (*)
1. Open the store.
2. Open browser console.
3. Paste and execute the following code:
```javascript
define('my-module', {
  foo: 'bar'
});
require(['my-module', function(){});
```
4. Verify that no error similar to `TypeError: Cannot read property 'map' of null` is thrown.

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#28340: Fix handling of mixins for modules with no dependencies defined
